builder.go

```diff
@@ -134,7 +134,7 @@ func check(resources []Resource) error {
 				return err
 			}
 			if _, ok := cache[dep.FromResource]; !ok {
-				return fmt.Errorf("Dependent resource %s doesn't exist. Validation failure.", dep.FromResource)
+				return fmt.Errorf("Dependent resource %s doesn't exist", dep.FromResource)
```

Comment:
@melugoyal this check can easily be implemented using a negation of ErrorMapper type check. Lets not add more error types...

All resource based errors are collected as ErrorMapper types and returned to the caller.